### ISSUE TITLE
MD037/no-space-in-emphasis reported with formulas

### ISSUE THREAD

**Issue by stefan-muc** [DavidAnson/vscode-markdownlint#129]
MD037/no-space-in-emphasis gets reported on first list element:

```
# Test Document

$$foobar$$ (1)

This is some rubbish text that can also be left out

* foobar
* foobar line that can be left out
* $foobar$
```

Which will render as:
![Clipboard01](https://user-images.githubusercontent.com/544335/94936468-908f0c00-04ce-11eb-817e-2b0398f76cbf.png)

I'm using [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) in Visual Studio Code together with [Markdown+Math](https://marketplace.visualstudio.com/items?itemName=goessner.mdmath). Last one uses $ and $$ for inline formulas just like LATEX does it.
Markdownlit seems to be confused by it, because it reports fist list element as "MD037/no-space-in-emphasis" when surrounded by

* a full line formula before that list
* this formula having an formula number in brackets
* an inline formula later on in that document (not necessarily in that list)

**Comment by DavidAnson** [comment]
Simplified repro for VS Code with only `vscode-markdownlint` extension installed:

```md
# Issue 129

$$math$$ text

* text
* text
```

**Comment by DavidAnson** [comment]
The presence of text after the math block (`$$math$$ text`) causes the https://www.npmjs.com/package/@iktakahiro/markdown-it-katex plugin referenced above to think the math block never ended, so the entire body of the file is believed to be inside the math block. We could discuss whether or not emphasis is meaningful in a math block, as in the following example, but the  bigger problem is the unwanted parsing.

```md
# Issue 129

$$

* text
* text

$$
```

The extension you reference seems to use the https://github.com/goessner/markdown-it-texmath library which explicitly supports the "display + equation number" syntax you're using above. Switching the `vscode-markdownlint` extension to use that plugin  resolves the issue you're reporting (due to it recognizing the math block as expected) but does report an issue with my example  (the emphasis rule doesn't know not to look for emphasis in a math block). (This could be addressed with an inline comment to disable that rule - it's a better place to be because the parsing is consistent with expectations.)

I will look into whether switching math parsers seems likely to break other/existing scenarios - if not, I'll do so for the next release of this extension.

**Comment by DavidAnson** [comment]
Here's the diff for future reference:

```diff
diff --git a/extension.js b/extension.js
index ed53cab..281a1d2 100644
--- a/extension.js
+++ b/extension.js
@@ -324,7 +324,7 @@ function markdownlintWrapper (name, text, config) {
 		config,
 		"customRules": getCustomRules(),
 		"handleRuleFailures": true,
-		"markdownItPlugins": [ [ require("@iktakahiro/markdown-it-katex") ] ],
+		"markdownItPlugins": [ [ require("markdown-it-texmath"), { engine: { renderToString: () => "" } } ] ],
 		"resultVersion": 3
 	};
 	let results = [];
diff --git a/package.json b/package.json
index ce4e7e1..735b3c1 100644
--- a/package.json
+++ b/package.json
@@ -31,9 +31,9 @@
 		"ignore": "5.1.8",
 		"js-yaml": "3.14.0",
 		"jsonc-parser": "2.3.1",
+		"markdown-it-texmath": "0.8.0",
 		"markdownlint": "0.21.0",
 		"markdownlint-rule-helpers": "0.12.0",
-		"@iktakahiro/markdown-it-katex": "4.0.0",
 		"minimatch": "3.0.4"
 	},
 	"devDependencies": {
```

**Comment by DavidAnson** [comment]
I've published version `0.37.1` of the extension with this change.

**Comment by stefan-muc** [comment]
Thank you very much, that was a very quick response and analysis! I can confirm that the new version fixes this bug.
